### PR TITLE
Attachments: uploading files, processing thumbnails, DB relations

### PR DESCRIPTION
**Pull request by clbn**

### CODE REVIEW

File: app/controllers/api/v1/AttachmentsController.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  }
+
+  AttachmentsController.create = function(req, res) {
+    if (!req.user)
+      return res.status(401).jsonp({ err: 'Not found' })
+
+    var form = new formidable.IncomingForm()
+
+    form.parse(req, function(err, fields, files) {
+      //console.log('fields', fields)
```
Comment (epicmonkey) [inline app/controllers/api/v1/AttachmentsController.js]: Let's drop commented out code unless there are reasons to keep it.

File: config/environments/development.js [hunk 2]
```diff
@@ -8,6 +8,17 @@ exports.getConfig = function() {
     saltSecret: 'secret token',
     secret: 'secret',
 
-    origin: 'http://localhost:3333'
+    origin: 'http://localhost:3333',
+
+    attachments: {
+      // Make sure that all directories here have a trailing slash
+      urlDir: 'http://frf.creagenics.com:3000/attachments/original/',
```
Comment (epicmonkey) [inline config/environments/development.js]: Can you change development settings to localhost please? That'd make it super easy to run server and html-client for newcomers.

**Comment by clbn** [comment]
In the commit update, I removed commented code and updated dev settings.

**Comment by epicmonkey** [comment]
Did you check why travis failed?

**Comment by clbn** [comment]
:-1: Don't merge this please.

Tests are fine now, but it seems that after a rebase (it's been a week of commits) all attachments are missing on page reload. I'm looking at it now.

**Comment by clbn** [comment]
Well, NOW both test and actual functionality in the browser are fine.

**Comment by epicmonkey** [comment]
Very cool! I'll review tonight.

File: app/models/attachment.js [hunk 3]
```diff
@@ -63,43 +71,114 @@ exports.addModel = function(database) {
...
+      var tmpPath = that.file.path
+      var originalPath = that.getPath()
+
+      fs.rename(tmpPath, originalPath, function() {
+        if ('image') { // TODO: support for various media types
+          gm(originalPath).size(function (err, size) {
+            // Check if we need to resize
+            if (size !== undefined && (size.width > 525 || size.height > 175)) {
+              // Looks big enough, needs a resize
+              that.noThumbnail = ''
```
Comment (epicmonkey) [inline app/models/attachment.js]: In the application we use '0' for false values and '1' for true ones. Let's follow this patterns unless there are strong objections.

File: config/environments/test.js [hunk 4]
```diff
@@ -1,14 +1,28 @@
...
     database: 3,
 
     saltSecret: 'secret token',
     secret: 'secret',
 
-    host: "http://localhost:31337",
     origin: 'http://localhost:3333'
   }
+
+  config.host = 'http://localhost:' + config.port
```
Comment (epicmonkey) [inline config/environments/test.js]: :+1:

File: app/models/attachment.js (hunk 3 above)
Comment (clbn) [inline app/models/attachment.js]: Well, the reason I chose `'true'` and `''` for this, is using `if (that.noThumbnail)` as it's a usual boolean. If we change them to `'1'` and `'0'`, we should explicitly use `if (that.noThumbnail === '1')` instead, which is quite ugly.

**Pull request merged**